camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/common/KafkaReader.java

```diff
@@ -163,53 +168,91 @@ public boolean fetch() throws IOException {
     try {
       fetchResponse = simpleConsumer.fetch(fetchRequest);
       if (fetchResponse.hasError()) {
+        throw new RuntimeException();
+      }
+      return processFetchResponse(fetchResponse, tempTime);
+    } catch (Exception e) {
+      log.info("Exception generated during fetch for topic " + kafkaRequest.getTopic()
+          + ". Will refresh topic metadata and retry.");
+      return refreshTopicMetadataAndRetryFetch(fetchRequest, tempTime);
+    }
+  }
+
+  private boolean refreshTopicMetadataAndRetryFetch(FetchRequest fetchRequest, long tempTime) {
+    try {
+      refreshTopicMetadata();
+      FetchResponse fetchResponse = simpleConsumer.fetch(fetchRequest);
+      if (fetchResponse.hasError()) {
         log.info("Error encountered during a fetch request from Kafka");
         log.info("Error Code generated : "
             + fetchResponse.errorCode(kafkaRequest.getTopic(), kafkaRequest.getPartition()));
         return false;
-      } else {
-        ByteBufferMessageSet messageBuffer =
-            fetchResponse.messageSet(kafkaRequest.getTopic(), kafkaRequest.getPartition());
-        lastFetchTime = (System.currentTimeMillis() - tempTime);
-        log.debug("Time taken to fetch : " + (lastFetchTime / 1000) + " seconds");
-        log.debug("The size of the ByteBufferMessageSet returned is : " + messageBuffer.sizeInBytes());
-        int skipped = 0;
-        totalFetchTime += lastFetchTime;
-        messageIter = messageBuffer.iterator();
-        //boolean flag = false;
-        Iterator<MessageAndOffset> messageIter2 = messageBuffer.iterator();
-        MessageAndOffset message = null;
-        while (messageIter2.hasNext()) {
-          message = messageIter2.next();
-          if (message.offset() < currentOffset) {
-            //flag = true;
-            skipped++;
-          } else {
-            log.debug("Skipped offsets till : " + message.offset());
-            break;
-          }
-        }
-        log.debug("Number of offsets to be skipped: " + skipped);
-        while (skipped != 0) {
-          MessageAndOffset skippedMessage = messageIter.next();
-          log.debug("Skipping offset : " + skippedMessage.offset());
-          skipped--;
-        }
+      }
+      return processFetchResponse(fetchResponse, tempTime);
+    } catch (Exception e) {
+      log.info("Exception generated during fetch for topic " + kafkaRequest.getTopic()
+          + ". This topic will be skipped.");
+      e.printStackTrace();
```

Comment:
Fixed. The printStackTrace() is in the old code, and there are other occurrences of printStackTrace() in the old code.